pkg/secret/factory.go

```diff
@@ -105,17 +106,24 @@ func (f *Factory) validate() error {
 		}
 	}
 
-	if f.GitUser != "" && !(strings.HasPrefix(f.GitUrl, "http://") || strings.HasPrefix(f.GitUrl, "https://")) {
-		return errors.Errorf("must provide a valid git url for basic auth (ex. https://github.com)")
+	if f.GitUser != "" {
+		if match, _ := regexp.Match(`^(?:https?://)(?:[a-zA-Z0-9\-\.]+(?::\d+)?)(?:\/(?:$|[^/]))?$`, []byte(f.GitUrl)); !match {
+			return errors.Errorf("must provide a valid git url without the repository path for basic auth (ex. https://github.com)")
+		}
 	}
 
-	if f.GitSshKeyFile != "" && !strings.HasPrefix(f.GitUrl, "git@") {
+	if f.GitSshKeyFile != "" && !validateGitSshAddress(f.GitUrl) {
 		return errors.Errorf("must provide a valid git url for SSH (ex. git@github.com)")
 	}
 
 	return nil
 }
 
+func validateGitSshAddress(address string) bool {
+	re := regexp.MustCompile(`^git@github\.com$`)
```

Comment:
We have to support domains other than `github.com` and the git implementation doesn't specify that the user is always `git`